src/admin/render_readmes.rs

```diff
@@ -184,26 +186,35 @@ fn render_pkg_readme<R: Read>(mut archive: Archive<R>, pkg_name: &str) -> anyhow
     };
 
     let rendered = {
-        let readme = manifest.package().readme();
-        if !readme.is_some() {
-            return Ok("".to_string());
-        }
-
-        let readme_path = readme.as_path().unwrap_or_else(|| Path::new("README.md"));
-        let path = Path::new(pkg_name).join(readme_path);
+        let readme = manifest
+            .package
+            .as_ref()
+            .and_then(|p| p.readme.as_ref())
+            .and_then(|r| r.as_ref().as_local());
+
+        let readme_path = match readme {
+            Some(StringOrBool::Bool(bool)) if !(*bool) => return Ok("".to_string()),
```

Comment:
I like that the `bool` handling is more explicit now, rather than being hidden in `OptionalFile::as_path()`.